copy-edit-topic-into-git (#449)
Umuzi-org/ACN-syllabus: content/topics/git/_index.md

title: Git Basics
---

## What's Git all about?

Git is an industry-standard tool used for collaboration within teams. We teach it quite early on in our course because without it you will be fairly useless as a professional developer. It's important. The Git book (below) has a better introduction.

## The Git Book

You don't need to know absolutely everything about Git to start using it. We keep our Git repositories on Github. Don't worry about the server setup stuff.

The first three chapters will serve you well initially:
https://git-scm.com/book/en/v2

## What we expect from you

As you progress through, you will have to write a lot of code. Your code will be submitted to us for review in the form of a Git repository. You will be expected to be comfortable using git from the command line.

## General best practices

- make a separate repo for every project (we don't want to have to sift through unrelated code when marking your work)

> - make sure that you _push_ your latest work before the end of the business day **every day**

Reviewer: ```suggestion
- make sure that you _push_ your latest work before logging off for the day **every day**
```
